### PR TITLE
Fireplace: Grayscale update and calibrator.

### PR DESCRIPTION
**Pull request by fuglaro**
* Launch calibrator by pressing A.
* Saves /thumbyGS.cfg for other games to use (needs updates).

### CODE REVIEW

File: Fireplace/thumbyGrayscale.py [hunk 1]
```diff
@@ -131,8 +123,65 @@ def setFrame(self, frame):
 _ST_THREAD       = const(0)
 _ST_COPY_BUFFS   = const(1)
 _ST_PENDING_CMD  = const(2)
-_ST_CONTRAST     = const(3)
-_ST_INVERT       = const(4)
+_ST_INVERT       = const(3)
+# Various display devices will run at naturally different clock
+# frequencies. This paramater allows for varyiance by adjusting the
```
Comment (TPReal) [inline Fireplace/thumbyGrayscale.py:128]: -> variance

File: Fireplace/thumbyGrayscale.py [hunk 2]
```diff
@@ -131,8 +123,65 @@ def setFrame(self, frame):
...
+# Newer models of the Thumby which started appearing in late 2022
+# used a different OLED manufacturer, which has a subtly different
+# design, and which gives nuanced differences to behaviours around the
+# hardware trick used to keep the display in sync with each subframe.
+# Unfortunately, it has a much lower maximum nominal clock frequency
+# which causes a noticable strobing. Fortunately, it can trap
+# the row counter much more effectively, and can thus greatly reduce the
+# size of the offscreen capture area, along with the strobing.
+# 0: Basic - Standard 17 row capture area with full row capture per subframe,
+#    each subframe has it's own contrast value.
```
Comment (TPReal) [inline Fireplace/thumbyGrayscale.py:140]: it's -> its

File: Fireplace/thumbyGrayscale.py (hunk 1 above)
Comment (fuglaro) [reply Fireplace/thumbyGrayscale.py]: Thanks!

File: Fireplace/thumbyGrayscale.py (hunk 2 above)
Comment (fuglaro) [reply Fireplace/thumbyGrayscale.py]: Thank you!

**Pull request merged**